nifi-toolkit/nifi-toolkit-encrypt-config/pom.xml

```diff
@@ -167,10 +167,12 @@
                 <groupId>org.apache.rat</groupId>
                 <artifactId>apache-rat-plugin</artifactId>
                 <configuration>
+                    <consoleOutput>true</consoleOutput>
                     <excludes combine.children="append">
                         <exclude>src/test/resources/scrypt.py</exclude>
-                        <exclude>src/test/resources/secure_hash.key</exclude>
-                        <exclude>src/test/resources/secure_hash_128.key</exclude>
+                        <!-- use wildcard for below files as tests generate additional files during the build -->
+                        <exclude>**/secure_hash.key</exclude>
+                        <exclude>**/secure_hash_128.key</exclude>
```

Comment:
Thanks @ijokarumawak. That confirms what I and others have been able to gather. It looks like that file is created by the command line tool that is under test. I don't think there is a way to change its location, other than perhaps changing the working directory of the test prior to executing the tool. Reliably removing the file would be a good approach as well. 

We may also want to disable these tests to get master building reliably on all platforms and then re-enable them once we have sorted this out.

Thanks for jumping in to assist!

Reply:
@kevdoran Yeah, I looked at the source code and it doesn't seem to be a way to change output dir currently. I think it is not only a problem for test, but also it might be problematic in real usage. When I executed the tool, it generates `secure_hash.key` in current directory:
```
$ ./bin/encrypt-config.sh -n /tmp/enc-test/nifi.properties -o /tmp/enc-test/nifi-enc.properties -b /tmp/enc-test/bootstrap.conf --verbose
$ ll
total 60
drwxrwxr-x 6 nifi nifi  4096 Apr 20 02:42 ./
drwxrwxr-x 3 nifi nifi  4096 Apr 19 03:57 ../
drwxr-xr-x 2 nifi nifi  4096 Apr 19 01:45 bin/
drwxr-xr-x 3 nifi nifi  4096 Apr 19 01:45 classpath/
drwxr-xr-x 2 nifi nifi  4096 Apr 19 01:45 conf/
drwxrwxr-x 2 nifi nifi 12288 Apr 19 03:57 lib/
-rw-r--r-- 1 nifi nifi 15986 Apr 19 01:45 LICENSE
-rw-r--r-- 1 nifi nifi  5473 Apr 19 01:45 NOTICE
-rw------- 1 nifi nifi    91 Apr 20 02:42 secure_hash.key
```

Since the key is baked into the bootstrap.conf, the secure_hash.key is not needed to be written as a file I guess. I'm new to this tool, so I can be wrong.

Reply:
Thanks for taking a look! I don't know enough about the tool to make a change to this behavior as it might have good reason it needs to work that way. For now I will update the PR to disable the tests, and we can take our time to discuss the best approach that looks at the tests and tool holistically.

Reply:
@kevdoran Sounds good. I will update NIFI-5100 to summarize this conversation so that we can make the tool better. Thanks!

Reply:
Ok, here's what I've got so far for the secure_hash.key file issue. It does not look trivial to find and disable the tests that could be creating this file, as I think it can come from both the existing (pre-NIFI-4942) and the recently introduced test cases.

The path is hardcoded to ./secure_hash.key, but the test class does appear to try to account for that here: https://github.com/kevdoran/nifi/blob/master/nifi-toolkit/nifi-toolkit-encrypt-config/src/test/groov...

I'm not sure why that is not working on some Linux platforms. I'm wondering if it has to do with the fact that it is a static field and how the classes get loaded for these test cases, but in any case it means that any execution of the tool could create the file that we want to avoid.

So options I see are:

- leave the tests enabled, leave the wildcard directory for the RAT exclusions, and maybe add the file to a .gitignore listing so as to prevent it from getting accidentally committed
- try to find every test that is creating this file and temporarily disable it (non-trivial from what I can tell, but maybe I am missing something obvious)
- disable the entire ConfigEncryptionToolTest suite (for now)

I think all of these approaches only address the immediate issues and still require a longer-term solution.